pandas/core/tools/timedeltas.py

```diff
@@ -118,8 +123,18 @@ def to_timedelta(arg, unit=None, errors="raise"):
             "arg must be a string, timedelta, list, tuple, 1-d array, or Series"
         )
 
-    if isinstance(arg, str) and unit is not None:
-        raise ValueError("unit must not be specified if the input is/contains a str")
+    if isinstance(arg, str):
+        if unit is not None:
+            raise ValueError(
+                "unit must not be specified if the input is/contains a str"
+            )
```

Comment:
hmm we *already* have this check on L103, so something must be going wrong here.

Reply:
if you do: `pd.to_timedelta("1 Y")` then unit is not specified and by default None. So then you never reach this code.